x-pack/plugins/siem/server/lib/events/elasticsearch_adapter.ts

```diff
@@ -56,10 +56,13 @@ export class ElasticsearchEventsAdapter implements EventsAdapter {
   constructor(private readonly framework: FrameworkAdapter) {}
 
   public async getEvents(request: FrameworkRequest, options: RequestOptions): Promise<EventsData> {
+    // Adding @timestamp here as gql doesn't support fields prefixed w/ @ and we've removed the ECS field mappings from query.dsl.ts
```

Comment:
I think what you want is to use the reduceFields function where it will only include fields that were mentioned in the GraphQL fields section as well as re-map things such as `@timestamp` to be `timestamp`. Otherwise this is going to have the extra field of `timestamp` and later if you re-use the query and don't want timestamp to be included in the output you can.

So, this should work for you:

```ts
const queryOptions = cloneDeep(options);
queryOptions.fields = reduceFields(options.fields, eventFieldsMap);
```

I would remove the extra comment because it doesn't apply after updating to this pattern.

In order for the build to pass we need to update the file `x-pack/plugins/siem/server/lib/framework/types.ts` to have a readonly array like so:

```ts
export interface RequestOptions extends RequestBasicOptions {
  pagination: PaginationInput;
  fields: ReadonlyArray<string>;
  sortField?: SortField;
}
```